# Corpus exploration — photographed prescriptions

This notebook reports the immutable EDA output generated by `scripts/analyze_corpus.py` over all 310 images and final OCR JSONL. Raw pages contain health information and are not displayed here.

In [1]:
import json
from pathlib import Path
root = Path.cwd().parent if Path.cwd().name == 'notebooks' else Path.cwd()
eda = json.loads((root/'reports/evidence/corpus_eda.json').read_text())
print('pages:', eda['pages'])
print('bytes:', eda['bytes'])
print('median dimensions:', eda['width_px']['median'], 'x', eda['height_px']['median'])
print('portrait fraction:', round(eda['portrait_fraction'], 4))
print('non-empty OCR pages:', eda['ocr_nonempty_pages'])
print('non-overlapped OCR words:', eda['ocr_word_occurrences_no_overlap'])
print('mean OCR words/page:', round(eda['mean_ocr_words_per_page'], 2))

pages: 310
bytes: 84388670
median dimensions: 1067 x 1326
portrait fraction: 0.9484
non-empty OCR pages: 310
non-overlapped OCR words: 42642
mean OCR words/page: 137.55


## Findings

- 94.8% of pages are portrait, but dimensions range from 593–1434 px wide and 672–1536 px high.
- Median brightness is 154.85/255; contrast ranges widely (standard deviation 13.94–81.01), supporting an enhancement ablation.
- The corpus mixes printed templates with handwritten Bangla/English medicine lines, shadows, perspective and Bengali numerals.
- All 310 pages yielded non-empty OCR, but 42,642 usable word occurrences remain below the required 60,000-word floor. Overlap is not counted toward corpus size.
- Twelve pages are held out with 61 medication labels. Raw images and OCR must remain access-controlled because they contain health information.